### PR TITLE
update README

### PR DESCRIPTION
**Pull request by mahmoud-a-ali**
- update readme
- change vrep_common pkg to coppeliasim_msgs_srvs

### CODE REVIEW

File: CMakeLists.txt
Comment (gavanderhoorn) [inline CMakeLists.txt:31]: Is there some tool (from CoppeliaSim perhaps) which we could use to get this automatically created?

File: CMakeLists.txt
Comment (gavanderhoorn) [inline CMakeLists.txt]:
```suggestion
  coppeliasim_msgs_srvs
  image_transport
  roscpp
  tf
```

(always sort alphabetically and list pkgs one-per-line)

File: README.md
Comment (gavanderhoorn) [inline README.md]:
```suggestion
A collection of ROS services and publishers wrapping the [CoppeliaSim APIs][]. These services and publishers use the `.msg`s and `.srv`s files defined in the [coppeliasim_msgs_srvs][] package. This package enables ROS to control the simulation inside CoppeliaSim (e.g. start/stop the simulation, loading/unloading scenes/modules, ... etc), and it allows CoppeliaSim to publish important information about the simulation (e.g. clock, simulation_state, time_step, simulation_time) on ROS topics.
```

File: README.md
Comment (gavanderhoorn) [inline README.md]:
```suggestion
The following instructions assume that a Catkin workspace has been created at `$HOME/catkin_ws` and a CoppeliaSim directory is located in the directory `$HOME/CoppeliaSim`. You always can update the paths based on your machine setup.
```

File: README.md
Comment (gavanderhoorn) [inline README.md]:
```suggestion
git clone https://github.com/tud-cor/coppeliasim_ros_services
```

File: README.md
Comment (gavanderhoorn) [inline README.md:17]:
```suggestion
# change to the root of the workspace
```

File: README.md
Comment (gavanderhoorn) [inline README.md:24]:
```suggestion
ls devel/lib | grep libsimExtRosServices.so
```

File: README.md
Comment (gavanderhoorn) [inline README.md:28]:
```suggestion
The generated plugin should be loaded while starting CoppeliaSim, this can be done by copying the file `libsimExtRosServices.so` to the main directory of CoppeliaSim (ie: `$HOME/CoppeliaSim`):
```

File: README.md
Comment (gavanderhoorn) [inline README.md:33]:
```suggestion
In a new terminal start a ROS Master before you start CoppelliaSim:
```

File: README.md
Comment (gavanderhoorn) [inline README.md:37]:
```suggestion
In another terminal run CoppeliaSim:
```

File: README.md
Comment (gavanderhoorn) [inline README.md:39]:
```suggestion
# change to CoppeliaSim dir 
```

File: README.md
Comment (gavanderhoorn) [inline README.md:42]:
```suggestion
# run CoppeliaSim
```

File: README.md
Comment (gavanderhoorn) [inline README.md:45]:
```suggestion
or you can launch CoppeliaSim using the [coppeliasim_run][] package:
```

File: README.md
Comment (gavanderhoorn) [inline README.md:50]:
```suggestion
In the same terminal running CoppeliaSim, you can see the following message indicating a successful load of the `RosServices` plugin:
```

File: README.md
Comment (gavanderhoorn) [inline README.md:55]:
```suggestion
In a new terminal, you can check all the available services and topics offered and published by the plugin:
```

File: README.md
Comment (gavanderhoorn) [inline README.md:56]:
```suggestion
- services: all services offered by the plugin will have `/coppeliasim_ros_services/` as a prefix
```

File: README.md
Comment (gavanderhoorn) [inline README.md:60]:
```suggestion
- topics: the two main topics published by the plugin are `/clock` and `/info`
```

File: README.md
Comment (gavanderhoorn) [inline README.md:66]:
```suggestion
The plugin publishes to two topics (`/clock` and `/info`), and offers many services. Here is an example of how you can use some of these services:
```

File: README.md
Comment (gavanderhoorn) [inline README.md:69]:
```suggestion
# service_request is empty, service_response is an integer indicating successful operation 
```

File: README.md
Comment (gavanderhoorn) [inline README.md:73]:
```suggestion
- Loading a module: loading the [coppeliasim_ros_control][] module (file: `libsimExtRosControl.so`, name: `RosControl`)
```

File: README.md
Comment (gavanderhoorn) [inline README.md:76]: Is this missing the values for the `fileName` and `pluginName` fields?

File: README.md
Comment (gavanderhoorn) [inline README.md:79]:
```suggestion
For more information about all the services, check the [CoppeliaSim APIs][] page.
```

File: README.md
Comment (gavanderhoorn) [inline README.md:83]:
```suggestion
Refer to the [ur5_coppeliasim_roscontrol][] package to see how you can use the `RosServices` plugin to control the simulation of a UR5 robot in CoppeliaSim.
```

File: README.md
Comment (gavanderhoorn) [inline README.md:96]: Please change these to point to `tud-cor` instead.

File: README.md
Comment (gavanderhoorn) [inline README.md:101]:
```suggestion
[simLoadScene()]: https://www.coppeliarobotics.com/helpFiles/en/regularApi/simLoadScene.htm
```

File: README.md
Comment (gavanderhoorn) [inline README.md:102]:
```suggestion
[CoppeliaSim APIs]: https://www.coppeliarobotics.com/helpFiles/en/apiFunctionListCategory.htm
```

File: README.md
Comment (gavanderhoorn) [inline README.md:104]: Is this link used in this file? If not: perhaps remove?

File: include/simExtRosService/simExtRosServer.h
Comment (gavanderhoorn) [inline include/simExtRosService/simExtRosServer.h:29]: Same question as earlier: if this is normally auto-generated, could we make use of a auto-generated version? Or does that not exist for CoppeliaSim yet?

File: package.xml
Comment (gavanderhoorn) [inline package.xml]: This would be great! Is Marc going to help maintain this? Or is it part of the template `package.xml`?

File: package.xml
Comment (gavanderhoorn) [inline package.xml]: We typically don't repeat email addresses in `author` elements (the `maintainer` element already has it).

File: package.xml
Comment (gavanderhoorn) [inline package.xml]:
```suggestion
  <url>https://github.com/tud-cor/coppeliasim_ros_services</url>
```

File: package.xml
Comment (gavanderhoorn) [inline package.xml]:
```suggestion
  <build_depend>opencv2</build_depend>
```

**Review by gavanderhoorn — CHANGES_REQUESTED** [review]
Great change ! Many thanks. This clarifies things immensely.

I've left some in-line comments. Could you take a look?

File: CMakeLists.txt
Comment (mahmoud-a-ali) [reply CMakeLists.txt:31]: I contacted Marc@coppeliasim, but another researcher at coppeliasim replied that this repository is discontinued and he doesn't know how to re-generate them.

File: CMakeLists.txt
Comment (mahmoud-a-ali) [reply CMakeLists.txt]: Ok, well take this into consideration next time

File: include/simExtRosService/simExtRosServer.h
Comment (mahmoud-a-ali) [reply include/simExtRosService/simExtRosServer.h:29]: coppeliasim people replied they don't know how to re-generate these files again.

File: package.xml
Comment (mahmoud-a-ali) [reply package.xml]: no, he won't. I will delete this line

**Comment by gavanderhoorn** [comment]
@mahmoud-a-ali : did you intend to merge this? There are about 22 unresolved comment.

Did you click the *show more*? Github collapses reviews with a lot of comments.

File: README.md
Comment (mahmoud-a-ali) [reply README.md:76]: yes. I modified it

**Comment by mahmoud-a-ali** [comment]
yes, sorry I did not see the collapsed comments. I looked back through all the unsolved comments and fixed them.

**Comment by gavanderhoorn** [comment]
No problem.

Can we make sure to use additional PRs for fixes after a PR has been merged?

That way we can review everything before merging.

Let's try to avoid direct pushes to `master`.

**Pull request merged**
